pkg/application/types.go

```diff
@@ -0,0 +1,26 @@
+package application
+
+import (
+	"github.com/redhat-developer/odo/pkg/component"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Application
+type App struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              AppSpec `json:"spec,omitempty"`
+}
+
+// AppSpec holds all information about application
+type AppSpec struct {
+	Name       string                    `json:"applicationName,omitempty"`
+	Components []component.ComponentSpec `json:"components,omitempty"`
```

Comment:
Wouldn't be enough to just to reference a component by some identification (like a name) rather than embedding whole component spec?